horovod/common/operations.cc

```diff
@@ -1167,8 +1170,20 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
   // We will ask for multiple threads, so other libraries like mpi4py can
   // be used together with Horovod if multi-threaded MPI is installed.
   int provided;
-  MPI_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, &provided);
 
+  // Calls MPI_Init if horovod_global.auto_mpi_init is set otherwise prints warning
+  if(horovod_global.auto_mpi_init){
+    int is_mpi_initialized= 0;
+    MPI_Initialized(&is_mpi_initialized);
+    if(!is_mpi_initialized)
+      MPI_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, &provided);
+  }else{
+    MPI_Query_thread(&provided);
```

Comment:
Here, let's check that MPI has been initialized & provided thread level == MPI_THREAD_MULTIPLE.  If those are both true, we don't need a warning.  If they are false, we should provide a warning and say that process will now likely crash.